### ISSUE TITLE
MMJoystick: Mapping Gamepad Issues [SOLVED-FIXED IN LATEST VERSION]

### ISSUE THREAD

**Issue by Oroken** [RPCS3/rpcs3#3796]
+cannot mapping input of gamepad
+wrong work button: gamepad work but it not work right in MMJoystick mode:
For ex: button 4:X, button 2:circle; button 3:triggle; button 1 square.  it should be   button 1: X; button 2: circle; button 3: square button 4: triggle,...
i think you should change the default setting like the last build or fix the remapping in config, i can't not input gamepad key but it can be work ingame: so strange( not work right lol) 
![dsd](https://user-images.githubusercontent.com/34053443/33306551-99fb2920-d445-11e7-8653-33954a620bae.JPG)

![24169359_1020165694792366_155068374_o](https://user-images.githubusercontent.com/34053443/33306561-a7125674-d445-11e7-9e3a-0bf13201ae27.jpg)
https://guides.github.com/features/mastering-markdown/
![24167221_1020165671459035_767994502_o](https://user-images.githubusercontent.com/34053443/33306566-aa7cb048-d445-11e7-8812-857f3f9bb28e.jpg)

**Comment by Megamouse** [comment]
The defaults will not work on every controller obviously.

Does the mapping not work at all?
No triggers, buttons, sticks ?

**Comment by Megamouse** [comment]
Try again with the newest version

**Comment by Oky12** [comment]
my joystick is working. but when i change mapping/button. i cannot press the input button after click button, EX = i click button cross. then i press button my joystick. but not change anything only -U. but that is solved with i click left button cross and press button joystick together in same time. and...... the button is change. @Megamouse that only add a waitting input. so..... after click then press button on controller. and the input button is inserted...

**Comment by Oroken** [comment]
T.T sorry to say but it just the same T.T it's work but not corectly and can not map the button T.T

**Comment by Psycho-A** [comment]
Can confirm. I have Logitech F310 gamepad with DInput, and I can set any binding, because it sets U- immediately. Latest master build.

**Comment by Oky12** [comment]
Click and press button together in same time..

**Comment by Psycho-A** [comment]
@Oky12, thanks, it helps, but it's not normal and a bug anyway...

**Comment by Oroken** [comment]
@Oky12 thanks, haha strange, it's work!!!

**Comment by Oky12** [comment]
Your welcome. I hope this bug fixed...

**Comment by Megamouse** [comment]
try this

https://ci.appveyor.com/project/rpcs3/rpcs3/build/0.0.4-7067/artifacts

**Comment by Psycho-A** [comment]
In 0.0.4-7067 binding keys stopped working at all. If before I could hold gamepad button, and then click binding, now I just get "U-" everywhere except left analog stick.

**Comment by Megamouse** [comment]
This must be some exceptionally bad gamepad you have there

**Comment by Psycho-A** [comment]
@Megamouse, tried on Logitech F710. The same thing. Everywhere except RPCS3 both gamepdas works just fine. Moreover, the previous emulator behavior was bad but still worked.

**Comment by alex88510** [comment]
My Logitech RumblePad 2 stopped working after #3611 merged. 
None of the subsequent input related pull request fixed it. 
The symptoms are same as above. 
The right analog is constantly "moving to left", and unable to map any other button. 
During ingame, the camera keep spinning because of the right analog is constantly "triggered" ...
And all the face buttons(triangle, square, etc.) are wrongly mapped, but I can't changed it because of the right analog issue again...

**Comment by Oky12** [comment]
![screenshot 67](https://user-images.githubusercontent.com/30838114/33377614-b6e9f26e-d544-11e7-9cac-9a22fc841cd3.png)
same. i just only mapping analog. all button cannot be mapping

**Comment by spyropt** [comment]
i opened a new issue but im having this same problem so its better to continue in here

Im using latest rpcs3 0.0.4-7071 using two different dinput gamepads if i try to map a button i always get z+ on the second i get U-
if i use the defaults that works on my arcade machine it seems it gets stuck on pressing up for example playing street fighter 2 turbo hd the fighter is always jumping

**Comment by Megamouse** [comment]
Retry with this:
https://ci.appveyor.com/project/rpcs3/rpcs3/build/0.0.4-7080/artifacts

**Comment by Oky12** [comment]
i try that. i think is same but on analog is not need click and point analog. and when i clicking other button i see text blinking (look screenshot). that is mising on pad configuration. waiting time. but the time is too fast and only blinking -__-
![sequence 04 00_04_37_32 still001](https://user-images.githubusercontent.com/30838114/33420787-7f37a1ec-d5e2-11e7-8f57-61909083c10a.jpg)

**Comment by Megamouse** [comment]
I don't understand

**Comment by Oky12** [comment]
What is you not understand?

**Comment by Megamouse** [comment]
Can you try this:
https://ci.appveyor.com/project/rpcs3/rpcs3/build/0.0.4-7086/artifacts

there will be a new button that says: "Filter Noise"
Press it without touching your controller and it should ignore all U- input.
The log should say something like:
MMJOY Calibration: Added axis [ 9709 = U- ] to blacklist. Value = 255

Unless I did not fully understrand your problem.

**Comment by Oky12** [comment]
so "filter noise" function to eliminate input u- and activate like time to immediately enter input button. 
![screenshot 70](https://user-images.githubusercontent.com/30838114/33422975-0af28312-d5ea-11e7-9ef8-c2b9062c377a.png)

**Comment by Megamouse** [comment]
you can try with this build.
it works just like master

**Comment by alex88510** [comment]
Tried the latest build https://github.com/RPCS3/rpcs3/commit/403d82b29c04721d38dd9e9a2b07e95e45e46326, the "filter noise" thing does not work after you close the configuration dialog. It only serves as a temporary fix while mapping the buttons. 
During ingame, the right analog is still constantly triggered, ingame camera spinning nonstop.

**Comment by Oky12** [comment]
Try mapping manual for right n left ( right analog ). U- value maybe is null

**Comment by Megamouse** [comment]
I think you misunderstood the filter noise button.
It serves as a means to disable spamming input like u- in your case from the controller. This does not affect any ingame controls.

So what you wanna do is put your controller untouched on the desk and then press "filter noise".
Now all the spam will be ignored.
Now you can and have to remap all the buttons or at least those that were wrong (like every button with u-).
The u- will not trigger anything ingame since you shouldn't have it mapped to any button.

I strongly believe some old or cheap controllers or drivers just keep spamming random values.

There's a website called http5gamepad.com.
You can see there if any button keeps getting pressed. Maybe

**Comment by alex88510** [comment]
@Megamouse 
I believe you mean html5gamepad.com
Tried it, no button keeps getting pressed but it didn't detect my d-pad buttons.

**Comment by Oky12** [comment]
maybe you driver. try using driver from windows

**Comment by alex88510** [comment]
@Oky12 I am using driver from windows, windows 10 by the way. 
Anyway, since it is possible to map all the buttons now, I guess the problem solved?

**Comment by Megamouse** [comment]
If the problem only exists on these few controllers, then yes. At least from my perspective.

for the Logitech controllers  there seems to be a switch that enables XInput usage.
maybe try that one instead.

**Comment by spyropt** [comment]
I have tested the latest version and it seems to work ok :) i can map all the buttons with filter noise

**Comment by Megamouse** [comment]
I have a fix incoming that should make the button irrelevant

**Comment by Oky12** [comment]
@Megamouse whether the filter noise feature will continue to exist in the future?

**Comment by Megamouse** [comment]
Okay this should finally work even withou the filter button

https://ci.appveyor.com/project/rpcs3/rpcs3/build/0.0.4-7093/artifacts

**Comment by Oky12** [comment]
that working. now i can easy to setting my joystick. i tested with gamepadtest app for ps3 and all button and analog work.
if your want to try this link download : https://drive.google.com/open?id=1tUfVbkymuXVL92UnKsdM6LhmpBoczPBV

**Comment by Psycho-A** [comment]
> Okay this should finally work even without the filter button

Works well for me without Filter noise button!
Also I think there's needed to a little change the default button layout to This one, for example. It matches the most of standard DirectInput controllers while current layout in RPCS3 matches XInput layout that doesn't need for MMJoy mode.

![z](https://user-images.githubusercontent.com/1223276/33478408-3c416810-d69a-11e7-8026-02d5f210ba67.PNG)

**Comment by spyropt** [comment]
Just tested and its all ok without filter noise :)

**Comment by Megamouse** [comment]
this can be closed.
If something else comes up, make a seperate Issue

**Comment by Oroken** [comment]
Ok let me edit the title a little bit

**Comment by Megamouse** [comment]
just close it dang it

**Comment by lazenes** [comment]
Although I use my joystick without any problems in other applications, when I enter the game in RPSC3, my character walks to the left, although I get a calibration completed warning with the Filter Noise button from the keystrokes, the same thing happens again in the game.
<img width="881" alt="image" src="https://github.com/RPCS3/rpcs3/assets/18098025/ef0afc57-f5e5-4cc5-a61e-177efbf0758f">
I need help with the subject
